design-proposals/network-binding-plugin/network-binding-plugin.md

```diff
@@ -1162,4 +1208,53 @@ Libvirt configuration snippet:
 </interface>
 ```
 
-- The domain itself is created by the `virt-launcher`.
\ No newline at end of file
+- The domain itself is created by the `virt-launcher`.
+
+# Appendix G: Alternatives to plugin sidecar container resource specification
+
+1. The cluster admin could configure worst case CPU/Memory requests and/or limits on the KubeVirt CR:
+
+Pros:
+- Already implemented.
+
+Cons:
+- Applies uniformly on all hook sidecars and network binding plugins
+    - Worst case CPU/Memory requests/limits should be defined on the KubeVirt CR which maybe wasteful for a combination of hook sidecars and network binding plugins.
+    - Coarse level of control.
+    - Only supports CPU and memory requests and limits.
+
+2. Additional API for network binding plugin sidecar resource specification:
```

Comment:
The network binding plugin **sidecar** container resources could quickly become common challenge for plugin authors, delegating it to Kubevirt could make it easier for plugin authors.

Note that at the next commit, that address the compute container resource overhead, one of the suggestions extend the API in similar way, the compute container is hardcoded in the field name `computeContainerResources`.
I suggest avoid hardcoded container name in the filed name to imply for which container the spesifed resources required for,  for example:
```
binding:
        mynetbindingplugin:
          sidecarImage: quay.io/kubevirt/mynetbindingplugin
          containerResources:
            name: <compute / network binding plugin sidecar>                        
            resources:
              requests:
                cpu: 200m
                memory: 20Mi
               limits: 
                 ...
```
Here we see the new `containerResources` field, it has `name` and `resources`  fields.
1. `container` stands for which container the resource requirements should apply for.
     There should be subset of acceptable names "compute" and "sidecar".
     When `compute` is specified, the desired resources should be added to compute container, as overhead to the existing resources.
     When `sidecar` is spesifed, the desired resources should be defined in the plugin sidecar container spec.
2. `resources` represent the desired resources.

Since the binding API is alpha, another options is changing the API structure in a way it enable more options for  the sidecar container sped, for example:
```
binding:
        mynetbindingplugin:
          sidecar:
            image: quay.io/kubevirt/mynetbindingplugin
            resources:
              requests:
                cpu: 200m
                memory: 20Mi
```
Here we see the new `sidecar` field, it has `image` and `resources` fields:
1. `image` stands for the sidecar container image, it should replace the existing `sidecarImage` field.
2. `resources` represent the desired resources.

Reply:
The second option makes sense now, but I do not think it is worth breaking the API for it.
While it is Alpha, we know that it has already users and our interest is not to break them.

I prefer we keep the option to break backward compatibility only for stronger reasons.

Regarding the 1st option, I think it is problematic: It will allow one plugin to affect other containers in general. That is not a good API IMO.
I think it is better to limit the plugin control over its own container and in the compute one, that is it.

Reply:
> Regarding the 1st option, I think it is problematic: It will allow one plugin to affect other containers in general. That is not a good API IMO. I think it is better to limit the plugin control over its own container and in the compute one, that is it.

IIUC, what @ormergi meant is that this field would be limited to "compute" or "sidecar" options.

Reply:
Yep, the container names should be validated in a way they can refer to the binding sidecar and compute container only.

Reply:
That is still a less optimal API IMO. A good API hints the user the boundaries of what is possible and what is not. An enum-like option as suggested here fits cases where we suspect the options will extend/change. But if we know they are limited, especially to only 2, it does not make sense to go this path IMO.

If we had 5 options, then for the same of simplicity, it would make more sense. On the other hand, if we had 5 options, it is most likely it would potentially grow and then will really fit an enum.

We do not expect this to grow as we want to limit this to the `compute` and  self `binding-plugin` container with no option to extend it.